### DISCUSSION TITLE
Trigger payload is not allowed - Directus Condition Flow fails

### DISCUSSION THREAD

**Discussion by VISHNU-P-M** [directus/directus#14561] [Archive]
I created a collection "Test" and it has toggle field `approved`,
I need to check the condition on this toggle field update to true or false.
And my filter condition is
`{
    "$trigger": {
        "payload": {
            "approved": {
                "_eq": true
            }
        }
    }
}`

I got the exception like 
`"message": "\"$trigger.<event, payload, key, collection>\" is not allowed",`

how can I rewrite my condition filter for logging while the approved field is updated to true?

@azrikahar I'm using directus version ^9.5.0!

**Comment by VISHNU-P-M** [comment]
I upgrade directus version to 9.14.3 and I got the result as I expected at that moment,
Now while I updated any other field in the test collection, the condition returning approved toggle true when actually approved is null.

`Trigger payload:`: 
`{
  "event": "test.items.update",
  "payload": {
    "sysApproved": true
  },
  "keys": [
    "2"
  ],
  "collection": "test"
}`

condition: 
`"$trigger": {
      "payload": {
        "approved": {
          "_eq": true
        }
      }
    }`

payload actually has no approved field, but it's returning true!
@azrikahar Why it is returning true?

**Reply by azrikahar** [reply]
Thanks for trying this on the latest version! That does seem odd. Let's track this in the new issue as mentioned here https://github.com/directus/directus/issues/13855#issuecomment-1192373076 and we'll investigate from there 👍
